drivers/dai/intel/dmic/dmic.c

```diff
@@ -552,16 +552,16 @@ static void dai_dmic_start(struct dai_intel_dmic *dmic)
 	int start_fir;
 
 	/* enable port */
-	key = k_spin_lock(&dmic->lock);
 	LOG_DBG("dmic_start(), dai_index = %d", dmic->dai_config_params.dai_index);
-	dmic->startcount = 0;
 
 	/* Compute unmute ramp gain update coefficient. */
 	dmic->gain_coef = db2lin_fixed(LOGRAMP_CONST_TERM / dmic->unmute_time_ms);
 
 	/* Initial gain value, convert Q12.20 to Q2.30 */
 	dmic->gain = Q_SHIFT_LEFT(db2lin_fixed(LOGRAMP_START_DB), 20, 30);
 
+	key = k_spin_lock(&dmic->lock);
```

Comment:
it depends really on what this lock is protecting. Is it protecting only the `.startcount` field? Or can other `dmic` fields (`.gain` and `.gain_coef`) be modified asynchronously too? Since this is a "start" function, I suspect that at this time no other context can try to access the DMIC DAI, so maybe you don't even need to protect the `.startcount`. Or is this protecting against DMIC interrupts triggering during initialisation? But in fact I agree with the comment in the commit message, that "this isn't fixing any existing issue"

Reply:
if you protect the `startcount` then you should protect the gain as well, they both used in dai_dmic_gain_ramp() under lock held and then probably the gain_coef as well.

It might be that protecting these variables are not needed at all...